File: android/src/main/kotlin/com/example/mango_player/platform/EventChannelHandler.kt
package com.example.mango_player.platform

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel
import com.example.mango_player.ijkplayer.IJKPlayerManager

class EventChannelHandler(
    private val messenger: BinaryMessenger,
    private val playerManager: IJKPlayerManager
) : EventChannel.StreamHandler {

    private var channel: EventChannel? = null
    private var eventSink: EventChannel.EventSink? = null

    fun startListening() {
        channel = EventChannel(messenger, "com.mangoplayer/events")
        channel?.setStreamHandler(this)
        playerManager.setEventSink(eventSink)
    }

    fun stopListening() {
        channel?.setStreamHandler(null)
        channel = null
        eventSink = null
        playerManager.setEventSink(null)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
        playerManager.setEventSink(events)
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
        playerManager.setEventSink(null)
    }
}

File: ios/Classes/Platform/EventChannelHandler.swift
import Flutter
import Foundation

class EventChannelHandler: NSObject, FlutterStreamHandler {
    private var channel: FlutterEventChannel?
    private let messenger: FlutterBinaryMessenger
    private let playerManager: IJKPlayerManager
    private var eventSink: FlutterEventSink?
    
    init(messenger: FlutterBinaryMessenger, playerManager: IJKPlayerManager) {
        self.messenger = messenger
        self.playerManager = playerManager
    }
    
    func startListening() {
        channel = FlutterEventChannel(name: "com.mangoplayer/events", binaryMessenger: messenger)
        channel?.setStreamHandler(self)
    }
    
    func stopListening() {
        channel?.setStreamHandler(nil)
        channel = nil
        eventSink = nil
        playerManager.setEventSink(sink: nil)
    }
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        playerManager.setEventSink(sink: events)
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        playerManager.setEventSink(sink: nil)
        return nil
    }
}
